modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartQuickAdd.js

```diff
@@ -0,0 +1,204 @@
+	'/o/headless-commerce-delivery-catalog/v1.0/channels';
+
+const channelsApiURL = new URL(
+	`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}`,
+	themeDisplay.getPortalURL()
+);
+
+const formattedProducts = [];
+
+const ProductAutocompleteList = ({onItemClick, sourceItems}) => (
+	<ClayDropDown.ItemList>
+		{sourceItems.map((product) => {
+			const {id, label, value} = product;
+
+			return (
+				<ClayDropDown.Item
+					key={id}
+					onClick={() => onItemClick(product)}
+				>
+					<div className="autofit-row autofit-row-center">
+						<div className="autofit-col mr-3 w-25">{value}</div>
+
+						<span className="ml-2 text-truncate-inline">
+							<span className="text-truncate">{label}</span>
+						</span>
+					</div>
+				</ClayDropDown.Item>
+			);
+		})}
+	</ClayDropDown.ItemList>
+);
+
+export default function CartQuickAdd() {
+	const {cartState, setCartState} = useContext(MiniCartContext);
+
+	const [productsQuery, setProductsQuery] = useState('');
+	const [quickAddToCartError, setQuickAddToCartError] = useState(false);
+	const [selectedProducts, setSelectedProducts] = useState([]);
+	const [products, setProducts] = useState();
+
+	const channelIdRef = useRef();
+
+	const {cartItems = []} = cartState;
+
+	useEffect(() => {
+		async function fetchProducts() {
+			const channelsResponse = await fetch(channelsApiURL.toString());
+
+			const channels = await channelsResponse.json();
+
+			channelIdRef.current = channels.items[0].id;
+
+			const productsApiURL = new URL(
+				`${themeDisplay.getPathContext()}${CHANNEL_RESOURCE_ENDPOINT}/${
+					channelIdRef.current
+				}/products?nestedFields=skus`,
+				themeDisplay.getPortalURL()
+			);
+
+			const productsResponse = await fetch(productsApiURL.toString());
+
+			const availableProducts = await productsResponse.json();
+
+			availableProducts.items.forEach((product) => {
+				const {name, skus} = product;
+
+				formattedProducts.push({
+					...product,
+					label: name,
+					value: skus[0].sku,
+				});
+			});
+
+			setProducts(availableProducts.items);
+		}
+
+		fetchProducts();
+	}, []);
+
+	const handleAddToCartClick = () => {
+		const itemSKUs = selectedProducts.map((item) => item.value);
+		const readyProducts = [];
+
+		products.forEach((product) => {
+			if (itemSKUs.includes(product.skus[0].sku)) {
+				const {productConfiguration, skus, urls} = product;
+
+				readyProducts.push({
+					...product,
+					price: skus[0].price,
+					productURLs: urls,
+					quantity: productConfiguration.minOrderQuantity,
+					settings: productConfiguration,
+					sku: skus[0].sku,
+					skuId: skus[0].id,
+				});
+			}
+		});
+
+		setCartState((cartState) => ({
+			...cartState,
+			cartItems: cartItems.concat(readyProducts),
+		}));
+
+		addToCart(
+			readyProducts,
+			cartState.id,
+			channelIdRef,
+			cartState.accountId
+		);
+
+		setSelectedProducts([]);
+	};
+
+	return (
+		<ClayForm.Group
+			className={classNames('p-3', quickAddToCartError && 'has-error')}
```

Comment:
I think this is the preferred syntax for classNames
```suggestion
			className={classNames('p-3', {'has-error': quickAddToCartError})}
```